sentry-servlet-jakarta/src/test/kotlin/io/sentry/servlet/jakarta/SentryRequestHttpServletRequestProcessorTest.kt

```diff
@@ -0,0 +1,163 @@
+        val eventRequest = event.request!!
+        assertEquals("GET", eventRequest.method)
+        assertEquals(
+            mapOf(
+                "some-header" to "some-header value",
+                "Accept" to "application/json"
+            ),
+            eventRequest.headers
+        )
+        assertEquals("http://example.com", eventRequest.url)
+        assertEquals("param1=xyz", eventRequest.queryString)
+    }
+
+    @Test
+    fun `attaches header with multiple values`() {
+        val request = mockRequest(
+            url = "http://example.com?param1=xyz",
+            headers = mapOf(
+                "another-header" to listOf("another value", "another value2")
+            )
+        )
+        val eventProcessor = SentryRequestHttpServletRequestProcessor(request)
+        val event = SentryEvent()
+
+        eventProcessor.process(event, null)
+
+        assertNotNull(event.request) {
+            assertEquals(
+                mapOf(
+                    "another-header" to "another value,another value2"
+                ),
+                it.headers
+            )
+        }
+    }
+
+    @Test
+    fun `does not attach cookies`() {
+        val request = mockRequest(
+            url = "http://example.com?param1=xyz",
+            headers = mapOf(
+                "Cookie" to "name=value"
+            )
+        )
+        val sentryOptions = SentryOptions()
+        sentryOptions.isSendDefaultPii = false
+        val eventProcessor = SentryRequestHttpServletRequestProcessor(request)
+        val event = SentryEvent()
+
+        eventProcessor.process(event, null)
+
+        assertNotNull(event.request) {
+            assertNull(it.cookies)
+        }
+    }
+
+    @Test
+    fun `does not attach sensitive headers`() {
+        val request = mockRequest(
+            url = "http://example.com?param1=xyz",
+            headers = mapOf(
+                "some-header" to "some-header value",
+                "X-FORWARDED-FOR" to "192.168.0.1",
+                "authorization" to "Token",
+                "Authorization" to "Token",
+                "Cookie" to "some cookies"
+            )
+        )
+        val sentryOptions = SentryOptions()
+        sentryOptions.isSendDefaultPii = false
+        val eventProcessor = SentryRequestHttpServletRequestProcessor(request)
+        val event = SentryEvent()
+
+        eventProcessor.process(event, null)
+
+        assertNotNull(event.request) { req ->
+            assertNotNull(req.headers) {
+                assertFalse(it.containsKey("X-FORWARDED-FOR"))
+                assertFalse(it.containsKey("Authorization"))
+                assertFalse(it.containsKey("authorization"))
+                assertFalse(it.containsKey("Cookies"))
+                assertTrue(it.containsKey("some-header"))
+            }
+        }
+    }
+}
+
+fun mockRequest(url: String, method: String = "GET", headers: Map<String, Any?> = emptyMap()): HttpServletRequest {
```

Comment:
This is a minimalistic replacement for `MockHttpServletRequestBuilder` and `MockHttpServletRequest` from `spring-test` which I didn't want to pull in to avoid Java 17.